bittensor/utils/balance.py

```diff
@@ -1,33 +1,46 @@
-""" Represent bittensor balance of the wallet with rao and tao
-"""
 # The MIT License (MIT)
 # Copyright © 2021 Yuma Rao
 
-# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
-# documentation files (the “Software”), to deal in the Software without restriction, including without limitation 
-# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, 
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the “Software”), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
 # and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 
-# The above copyright notice and this permission notice shall be included in all copies or substantial portions of 
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of
 # the Software.
 
 # THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
-# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
-# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION 
-# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 # DEALINGS IN THE SOFTWARE.
+from typing import Union
+
 
 class Balance:
-    """ Represent bittensor balance of the wallet with rao and tao
     """
-    unit = "\u03C4"
-    rao : int
+    Represents the bittensor balance of the wallet, stored as rao (int)
+    The Balance object is immutable, and can be used as a number or as a string
+    Can only guarantee that the balance is accurate to 9 decimal places (tao)
+    """
+
+    unit: str = "\u03C4" # This is the tao unit
+    rao_unit: str = "\u03C1" # This is the rao unit
+    rao: int
     tao: float
 
-    # Balance should always be an int and represent the satoshi of our token:rao
-    def __init__(self, balance):
-        self.rao = balance
-        self.tao = self.rao / pow(10, 9)
+    def __init__(self, balance: Union[int, float]):
+        if isinstance(balance, int):
```

Comment:
should we assert that `balance >= 0`?

Reply:
I would say it's okay to have a negative balance, for the sake of using bittensor as an api in other projects, treating Balance more like a number is good for multiple use-cases (tao-debt, idk)

Reply:
Sure, that makes sense to me